depthai_nodes/ml/parsers/xfeat.py

```diff
@@ -0,0 +1,527 @@
+    **ValueError**: If the output layer containing heatmaps is not specified.
+    """
+
+    def __init__(
+        self,
+        output_layer_feats: str = "",
+        output_layer_keypoints: str = "",
+        output_layer_heatmaps: str = "",
+        original_size: Tuple[float, float] = None,
+        input_size: Tuple[float, float] = (640, 352),
+        max_keypoints: int = 4096,
+    ):
+        """Initializes the XFeatBaseParser node."""
+        super().__init__()
+        self._target_input = self.createInput()  # used in stereo mode
+
+        self.output_layer_feats = output_layer_feats
+        self.output_layer_keypoints = output_layer_keypoints
+        self.output_layer_heatmaps = output_layer_heatmaps
+        self.original_size = original_size
+        self.input_size = input_size
+        self.max_keypoints = max_keypoints
+
+    @property
+    def reference_input(self) -> Optional[dai.Node.Input]:
+        """Returns the reference input."""
+        return self.input
+
+    @property
+    def target_input(self) -> Optional[dai.Node.Input]:
+        """Returns the target input."""
+        return self._target_input
+
+    @reference_input.setter
+    def reference_input(self, reference_input: Optional[dai.Node.Input]):
+        """Sets the reference input."""
+        self.input = reference_input
+
+    @target_input.setter
+    def target_input(self, target_input: Optional[dai.Node.Input]):
+        """Sets the target input."""
+        self._target_input = target_input
+
+    def build(
+        self,
+        head_config: Dict[str, Any],
+    ):
+        """Sets the head configuration for the parser.
+
+        Attributes
+        ----------
+        head_config : Dict
+            The head configuration for the parser.
+
+        Returns
+        -------
+        XFeatBaseParser
+            Returns the parser object with the head configuration set.
+        """
+
+        output_layers = head_config["outputs"]
+        if len(output_layers) != 3:
+            raise ValueError(
+                f"Only three output layers supported for XFeat, got {len(output_layers)} layers."
+            )
+
+        for layer in output_layers:
+            if "feats" in layer:
+                self.output_layer_feats = layer
+            elif "keypoints" in layer:
+                self.output_layer_keypoints = layer
+            elif "heatmaps" in layer:
+                self.output_layer_heatmaps = layer
+
+        self.original_size = head_config.get("original_size", self.original_size)
+        self.input_size = head_config.get("input_size", self.input_size)
+        self.max_keypoints = head_config.get("max_keypoints", self.max_keypoints)
+
+        return self
+
+    def setOutputLayerFeats(self, output_layer_feats):
+        """Sets the output layer containing features.
+
+        @param output_layer_feats: Name of the output layer containing features.
+        @type output_layer_feats: str
+        """
```

Comment:
should type checking be added to the setters?

Reply:
Added in [a87701f](https://github.com/luxonis/depthai-nodes/pull/104/commits/a87701f03b1f6b6e83ec4e3c4b12dd124e3cb1cf). Note that we should do this for all parsers.